string_registers.c

```diff
@@ -0,0 +1,119 @@
+/*
+  string_registers.c - get/set string register value by id or name
+
+  Part of grblHAL
+
+  Copyright (c) 2024-2025 Stig-Rune Skansgård
+
+  grblHAL is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  grblHAL is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with grblHAL. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "hal.h"
+
+#if STRING_REGISTERS_ENABLE
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "system.h"
+#include "settings.h"
+#include "ngc_params.h"
+#include "string_registers.h"
+
+#ifndef MAX_SR_LENGTH
+#define MAX_SR_LENGTH 50
+#endif
+typedef struct string_register {
+    string_register_id_t id;
+    char value[MAX_SR_LENGTH + 1];
+    struct string_register *next;
+} string_register_t;
+
+static string_register_t *string_registers = NULL;
+
+string_register_t *find_string_register_with_last (string_register_id_t id, string_register_t **last_register) {
+    string_register_t *current_register = string_registers;
+    int i = 0;
+
+    while(current_register != NULL) {
+
+        if(current_register->id == id) {
+            return current_register;
+
+        } else if (i++ < 1000) {
```

Comment:
This is an arbitrary limitation that I see no reason to have.

Reply:
this is/was a "safety-feature" I added as I was having some issue where the while-loop got stuck forever 😅 Should be good to remove now 👍